contrib/format-excel/src/main/java/org/apache/drill/exec/store/excel/ExcelBatchReader.java

```diff
@@ -519,7 +519,10 @@ public void load(Cell cell) {
       if (cellValue == null) {
         columnWriter.setNull();
       } else {
-        Instant timeStamp = new Instant(cellValue.getNumberValue());
+        logger.debug("Cell value: {}", cellValue.getNumberValue());
+
+        Date dt = DateUtil.getJavaDate(cellValue.getNumberValue());
```

Comment:
Is it possible to use either `java.util` or `org.joda.time` classes?

Reply:
That was actually the source of the bug.  I didn't realize it when I wrote the plugin but as it turns out, Excel generates all kinds of weird representations of dates.  Specifically in the case dates (no time) it generates a `double` and when it has a time, it generates a `long`.  Using the `DateUtil` class that is part of the POI library solves this problem and gets consistent dates out of Excel.

Reply:
Thanks for the explanation, I think we are good then.